src/DataGrid.tsx

```diff
@@ -487,21 +479,29 @@ function DataGrid<R, SR, K extends Key>(
   /**
    * callbacks
    */
-  const handleColumnResize = useCallback(
+  const handleColumnResize = useLatestFunc(
     (column: CalculatedColumn<R, SR>, width: number | 'max-content') => {
       if (width === 'max-content') {
-        setAutoResizeColumn(column);
+        flushSync(() => {
+          // TODO: only clear flex columns, if any?
+          setMeasuredColumnWidths(new Map());
+          setAutoResizeColumn(column);
+        });
         return;
       }
-      setColumnWidths((columnWidths) => {
-        const newColumnWidths = new Map(columnWidths);
-        newColumnWidths.set(column.key, width);
-        return newColumnWidths;
+
+      flushSync(() => {
```

Comment:
Any difference if `flushSync` is not used?

Reply:
It can make the bottom scroll bar flicker
![Recording 2022-09-21 at 13 28 26](https://user-images.githubusercontent.com/567105/191503714-62c40c30-98eb-4f42-941a-6ed30f79920f.gif)